### ISSUE TITLE
Regression: Composite models no longer render correctly

### ISSUE THREAD

**Issue by TBoshoven** [MinecraftForge/MinecraftForge#9347]
**Minecraft Version:** 1.19.3

**Forge Version:** 1.19.3-44.1.16

**Steps to Reproduce:**
 1. Open test mod `CompositeTest` (part of Forge)
 2. Give yourself a `composite_model_test:composite_block` (can grab from creative tab)
 3. Observe that it doesn't look right when held or placed in the world

**Description of issue:**
Composite models currently seem to be rendering incorrectly.
I haven't investigated very deeply (yet) but it looks like child models are being rotated, perhaps over the wrong axis.

[Back in 2020](https://github.com/MinecraftForge/MinecraftForge/pull/6650) I wrote this handy test mod to validate that composite models render correctly. It now shows that they do not:
![2023-02-16_23 09 43](https://user-images.githubusercontent.com/1949124/219548274-c0e33c47-2b72-4fa3-a416-d60e3c8620f4.png)
(the tiny blocks should overlap with the rendered outlines)

I also experienced this while attempting to update my own mod to 1.19.3.

**Comment by Shadows-of-Fire** [comment]
@gigaherz Since I _think_ you're the original author of composite model you may want to take a look at this

**Comment by TBoshoven** [comment]
I had a lot of trouble bisecting this so apologies for being a bit vague, but I'm pretty sure this is caused by https://github.com/MinecraftForge/MinecraftForge/pull/9065 .
Note: this makes it a regression, but introduced in 1.19.2 rather than 1.19.3 .

Verified by testing my own mod in Forge 43.1.55 (working) and 43.1.57 (not working) because I somehow had trouble running test mods on these commits.

**Comment by marchermans** [comment]
@amadornes Did we not fix this?

**Comment by TBoshoven** [comment]
We looked into this a bit (thanks @amadornes ).
It looks like composite models have nothing to do with this.
When reducing the model to one of its parts, the problem still exists.

It kind of looks like we simply got the transformation origin wrong, but we haven't quite figured out what the exact issue is and how it was introduced.
When simply scaling to origin, the "center" origin seems to map to one of the corners, and the corner origins map to a location outside the block.
